angu-fixed-header-table.js

```diff
@@ -1,82 +1,86 @@
 ﻿/**
+ * This code was initially based on:
+ *
  * AngularJS fixed header scrollable table directive
  * @author Jason Watmore <jason@pointblankdevelopment.com.au> (http://jasonwatmore.com)
- * @version 1.2.0
+ * @version 1.2.0 (https://github.com/cornflourblue/angu-fixed-header-table/commit/41f8dfd1f35a242f2931cbf1885c82cfaace671b)
  */
-(function () {
-    angular
-        .module('anguFixedHeaderTable', [])
-        .directive('fixedHeader', fixedHeader);
+(function(){
 
-    fixedHeader.$inject = ['$timeout', '$window'];
+    var Utils = (function(){
 
-    function debounce(func, wait, immediate) {
-        var timeout;
-        return function() {
-            var context = this, args = arguments;
-            var later = function() {
-                timeout = null;
-                if (!immediate) func.apply(context, args);
+        function debounce(func, wait, immediate){
+            var timeout;
+            return function() {
+                var context = this, args = arguments;
+                var later = function() {
+                    timeout = null;
+                    if (!immediate) func.apply(context, args);
+                };
+                var callNow = immediate && !timeout;
+                clearTimeout(timeout);
+                timeout = setTimeout(later, wait);
+                if (callNow) func.apply(context, args);
             };
-            var callNow = immediate && !timeout;
-            clearTimeout(timeout);
-            timeout = setTimeout(later, wait);
-            if (callNow) func.apply(context, args);
-        };
-    };
+        }
 
-    function fixedHeader($timeout, $window) {
+        function isInteger(strNum){
+            return /^\+?\d+$/.test(strNum);
```

Comment:
We could improve here to not create the regexp on each call since it is static?.